### PR TITLE
Remove unnecessary stuff from CMake

### PR DESCRIPTION
**Pull request by chfast**
- Remove boost::program_options requirement.
- Do not include gtest.
- Remove libcrypto requirement.

### CODE REVIEW

**Comment by howardwu** [comment]
@chfast nice updates! Could you sync your branch upstream and resolve conflicts? I'll merge the updates in

**Comment by chfast** [comment]
Yes, will do. Give me a bit of time.

**Comment by chfast** [comment]
Rebased.

File: libff/CMakeLists.txt [hunk 1]
```diff
@@ -55,11 +55,14 @@ target_link_libraries(
   ff
 
   ${GMP_LIBRARIES}
-  ${GMPXX_LIBRARIES}
-  ${CRYPTO_LIBRARIES}
   ${PROCPS_LIBRARIES}
   ${FF_EXTRALIBS}
 )
+target_include_directories(
```
Comment (howardwu) [inline libff/CMakeLists.txt]: What is the motivation for including over linking GMP? There's currently a linking error in this build: https://travis-ci.org/scipr-lab/libff/builds/269181457#L1262

Comment (chfast) [reply libff/CMakeLists.txt]: You need both (unless you use CMake's imported targets).
I think the issue is the variable `GMP_LIBRARIES` is never set. I will check it.

Comment (chfast) [reply libff/CMakeLists.txt]: I think the builds are failing because the GMP was previously pulled by additional libcrypto library.

**Comment by chfast** [comment]
Linking with GMP should be fixed now.

**Review by howardwu — APPROVED** [review]
LGTM

**Pull request merged**